src/app/api/chat/route.ts

```
@@ -0,0 +1,304 @@
+      return new Response('No messages provided', { status: 400 });
+    }
+    const query: string =
+      lastMessage.content || 
+      (lastMessage.parts as { type: string; text?: string }[])?.find(
+        (p) => p.type === "text",
+      )?.text || "";
+
+  // 1. Embed the query using HuggingFace (same model as ingest: all-MiniLM-L6-v2)
+  let embedding: number[] | null = null;
+
+  try {
```

ESLint

[error] Insert `··`

(prettier/prettier)